### ISSUE TITLE
[figma] Sync icons to include the new ones

### ISSUE THREAD

**Issue by oliviertassinari** [mui-org/mui-design-kits#45]
For example, https://mui.com/components/material-icons/?query=rocket&selected=Rocket is not available in the Figma icons set.

<img width="381" alt="Screenshot 2022-01-03 at 00 23 11" src="https://user-images.githubusercontent.com/3165635/147892109-51821bfa-b36d-47ac-9b64-81ce159efa68.png">

### Version

v5.0.1

### Context

https://mui.zendesk.com/agent/tickets/1964

**Comment by adrianmanea** [comment]
In the new update, we use [Google's ](https://www.figma.com/community/file/1035203688168086460) for the following reasons:
- [ ] consistent updates
- [ ] all icon styles available as variants (filled, outlined, sharp, rounded, two-tone)
- [ ] separate icon library which improves the performance of library greatly

**Comment by oliviertassinari** [comment]
@adrianmanea A couple of questions:

- What were we using before?
- Does it retain the Figma component name <> React component name direct mapping?
- Is Google keeping the Figma as up to date as https://fonts.google.com/icons?

I guess that what would be ideal is to flag the icons that are not natively available in `@mui/icons-material` but this outside of the scope of this issue and to add the few extra icons available in `@mui/icons-material` that are not in Google's icons.

**Comment by adrianmanea** [comment]
> What were we using before?

We were using a copy of the @mui/icons-material but being all styles in bulk we used node scripts to organize them. Somehow some icons did not go through as mentioned in this issue.

> Does it retain the Figma component name <> React component name direct mapping?

Nope, they use snake-case in the original file but if we make our own copy we can easily change to React component name. Doing this means we need to maintain our own community file. Happy to do so, not a lot of trouble since the icons are not frequently updated.

> Is Google keeping the Figma as up to date as https://fonts.google.com/icons?

I believe so, yes.

> flag the icons that are not natively available

Yep, we need to think of an automated process to flag these. So @mui/icons-material does not contain the whole icon set?

**Comment by oliviertassinari** [comment]
> Somehow some icons did not go through as mentioned in this issue.

I think that I have the answer to "somehow". Our Figma icons were released **before** the Rocket icon was introduced in React. Let's hope we don't have the same sync issue with Google's Figma file. 

> Nope, they use snake-case in the original file

I guess this depends on what will be the UX for designers/developers. If the snake-case doesn't make it harder to switch between the two environnement, then why not.

> @mui/icons-material does not contain the whole icon set

There are some that we skip, we might skip even more icons in the future, using a popularity metric to keep the package DX fast. The last source of misalignment is depending on when the icons are updated. Was this all done at the same time? Using the same source of truth?

**Comment by adrianmanea** [comment]
> The last source of misalignment is depending on when the icons are updated.

Indeed. Figma will be future proof from the icons standpoint but this might raise the number of issues on the core repo about missing icons? Or maybe not, people can always import them as custom SVGs.

> If the snake case doesn't make it harder to switch

I think the tradeoff is positive.

An alternative option would be to make a copy and maintain it ourselves, change the name to ComponentName and curate the icons so we won't have misalignments. Right now we are 2 days away from releasing the new update which solves all issues from this repo. Doing this alternative will postpone the update to one week at least - we need to verify each icon is present in the @mui/icons-material folder.

Happy to do the alternative as well.

**Comment by oliviertassinari** [comment]
Alright, happy to try what you are pushing for, see if it works for our Figma users.

---

For the future, I would personally advocate for the following "best" approach:

1. First and foremost, automate it all when it comes to icons.
1. `@mui/material-icons` is kept in sync with http://fonts.google.com/icons every 3 months. In React, we ignore the non-popular icons, we keep the legacy icons to not create breaking changes, we rename a few icons to be compatible with the React imports, we add extra highly popular icons (currently ~10 social icons, e.g. Twitter). Regarding progress, we are still exploring removing non-popular icons https://github.com/mui-org/material-ui/issues/29687. We don't have a bot to automatically sync the icons every quarter but wait on developers to complain which is not as efficient.
2. Figma icons are kept in sync with http://fonts.google.com/icons every 3 months, at the same frequency and date than `@mui/material-icons`. We use more or less the same import script, at least, have the key shared constants shared, e.g. how to name icons. However, compared to 2.: to not include the legacy icons, for the non-popular icons, flag them in a way that is clear that there are no React code imports, and add the same extra icons as available in React.



cc @michaldudak for context per https://www.notion.so/mui-org/Product-ownership-836c72feffcc4cb2b735b51527e6991a?p=2c123f5ccdb84f3b91c583463b620807.

**Comment by adrianmanea** [comment]
Ok cool, sounds like a plan. We will be watching the changes of the @mui/material-icons.

So I propose the following:
- Make a copy of Google's Figma Icons (of course, with the credits in place). [The license allows it.](https://creativecommons.org/licenses/by/4.0/)
- Rename components to match ours
- In the next update we delete the non-present icons from the @mui/material-icons

Waiting for confirmation 🚦

**Comment by oliviertassinari** [comment]
@adrianmanea Sounds good, for 3. I was proposing to label them, deleting them might be a bit too strong. If a designer really want it, he could ask his engineers to import the svg path directly, and combine it with `<SvgIcon>`.

**Comment by oliviertassinari** [comment]
I had a look at the new icons in v5.2.0. We didn't fix this issue, the icon Rocket is [available](https://mui.com/components/material-icons/?query=rocket) in `@mui/icons-material`, is [available](https://fonts.google.com/icons?icon.query=rocket) in fonts.google.com/icons, but is [not available](https://www.figma.com/community/file/1014241558898418245) in the community Figma file of Google.

**Comment by adrianmanea** [comment]
That's true, damn it. I searched for "rocket" and I assumed it was the "SportsTenis" icon.

Seems like we have to do the syncing process with the @mui/icons-material now. From what I can see, I need a script that:
- Moves all style flagged ("..._outlined") icons in separate folders
- Removes the style flag
- Converts to ComponentCase

>has to change the "style" prop each time 
That's also true.

**Comment by adrianmanea** [comment]
Ok, so we've updated the old libraries, e.g [Filled](https://www.figma.com/file/iUqeugKXVaB2diUqbNkgUF/Material-Design-Icons-v5.2.0-Filled?node-id=305%3A514).

Now we are 100% in sync with @mui/icons-material.

**Comment by oliviertassinari** [comment]
Great 👌
